autoPyTorch/evaluation/utils.py

```diff
@@ -0,0 +1,80 @@
+import queue
+from multiprocessing.queues import Queue
+from typing import List, Optional, Union
+
+import numpy as np
+
+from smac.runhistory.runhistory import RunValue
+
+__all__ = [
+    'read_queue',
+    'convert_multioutput_multiclass_to_multilabel',
+    'extract_learning_curve',
+    'empty_queue'
+]
+
+
+def read_queue(queue_: Queue) -> List[RunValue]:
+    stack: List[RunValue] = []
+    while True:
+        try:
+            rval: RunValue = queue_.get(timeout=1)
+        except queue.Empty:
+            break
+
+        # Check if there is a special placeholder value which tells us that
+        # we don't have to wait until the queue times out in order to
+        # retrieve the final value!
+        if 'final_queue_element' in rval:
+            del rval['final_queue_element']
+            do_break = True
+        else:
+            do_break = False
+        stack.append(rval)
+        if do_break:
+            break
+
+    if len(stack) == 0:
+        raise queue.Empty
+    else:
+        return stack
+
+
+def empty_queue(queue_: Queue) -> None:
+    while True:
+        try:
+            queue_.get(block=False)
+        except queue.Empty:
+            break
+
+    queue_.close()
+
+
+def extract_learning_curve(stack: List[RunValue], key: Optional[str] = None) -> List[List]:
+    learning_curve = []
+    for entry in stack:
+        if key:
```

Comment:
Seems like lazy coding for `if key is not None` to me :P

Reply:
fixed :P